scripts/editor/register-blocks.js

```diff
@@ -483,7 +483,14 @@ const registerVariation = (
 ) => {
 
 	// Append globalManifest data in to output.
-	blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	if (blockManifest['icon']) {
+		blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	} else {
+		// There is no icon passed to variation, use it's parent icon instead
+		import(`./../../blocks/init/src/Blocks/custom/${blockManifest.parentName}/manifest.json`).then((parentManifest) => {
```

Comment:
this is not the way to go because this path can change in the project and this setup needs to be as generic as possible.

You should pass additional key in the registerVariation function where you will get all blocks, and in that case you just find the icon from the paren key name:

https://github.com/infinum/eightshift-frontend-libs/blob/develop/blocks/init/src/Blocks/assets/scripts/application-blocks-editor.js